WebDriverAgentLib/Routing/FBSession.m

```diff
@@ -177,9 +178,14 @@ - (XCUIApplication *)activeApplication
     }
   }
 
-  NSString *defaultBundleId = [self.defaultActiveApplication isEqualToString:FBDefaultApplicationAuto]
+  NSString *defaultBundleId = isAuto
     ? nil
     : self.defaultActiveApplication;
+
+  if (nil != defaultBundleId && [self applicationStateWithBundleId:defaultBundleId] >= XCUIApplicationStateRunningForeground) {
```

Comment:
this check should be made at the start of this method
this setting has priority over testedApplication